### PR TITLE
UE edit of Razor file compilation doc

### LINKED ISSUE #5408: Compile Razor pages
aspnet/Razor#1809

### PR DESCRIPTION
**Pull request by scottaddie**
Fixes #5408

**Changes**
- Make the doc unavailable for ASP.NET Core 1.0, where precompilation isn't supported
- Replace tabs with versioning monikers
- Move an unused *.csproj* file from the *sample* folder to a *samples* folder for the metapackage doc using it
- Address Acrolinx suggestions, Gauntlet warnings, and Markdownlint warnings
- Add headings to improve readability
- Add an **Additional resources** section

**Internal Review Pages**
* [ASP.NET Core 1.1](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/views/view-compilation?view=aspnetcore-1.1&branch=pr-en-us-6464)
* [ASP.NET Core 2.0](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/views/view-compilation?view=aspnetcore-2.0&branch=pr-en-us-6464)
* [ASP.NET Core 2.1](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/views/view-compilation?view=aspnetcore-2.1&branch=pr-en-us-6464)

### CODE REVIEW

File: aspnetcore/mvc/views/view-compilation.md [hunk 1]
```diff
@@ -1,66 +1,92 @@
...
 ms.topic: article
 uid: mvc/views/view-compilation
 ---
-# Razor file (.cshtml) compilation in ASP.NET Core
+# Razor file compilation in ASP.NET Core
 
 By [Rick Anderson](https://twitter.com/RickAndMSFT)
 
-Razor views are compiled at runtime when the view is invoked. ASP.NET Core 2.1.0 or later compile views at build and publish time using [Razor Sdk](/aspnetcore/mvc/razor-pages/sdk). In ASP.NET Core 1.1, and ASP.NET Core 2.0, views can optionally be compiled at publish and deployed with the app &mdash; using the precompilation tool. 
+::: moniker range="= aspnetcore-1.1"
```
Comment (pranavkm) [inline aspnetcore/mvc/views/view-compilation.md]: Out of curiosity, do these appear like tabs? Should this in that case be ordered in descending (2.1 before 2.0)

File: aspnetcore/mvc/views/view-compilation.md [hunk 2]
```diff
@@ -1,66 +1,92 @@
...
-```xml
-<PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.ViewCompilation" Version="2.0.0" PrivateAssets="All" />
-```
+[!code-xml[](view-compilation/sample/DotNetFrameworkProject.csproj?name=snippet_ViewCompilationPackage)]
 
 If your project targets .NET Core, no changes are necessary.
 
-The ASP.NET Core 2.x project templates implicitly sets `MvcRazorCompileOnPublish` to `true` by default, which means this node can be safely removed from the *.csproj* file.
-    
+The ASP.NET Core 2.x project templates implicitly set the `MvcRazorCompileOnPublish` property to `true` by default. Consequently, this element can be safely removed from the *.csproj* file.
```
Comment (pranavkm) [inline aspnetcore/mvc/views/view-compilation.md]: Should we say we recommend removing the element from your csproj if it is set to `true`? Having the property will turn off Razor Sdk in 2.1, so it's less than desirable

**Comment by Rick-Anderson** [comment]
@pranavkm  asks
>Out of curiosity, do these appear like tabs?

See Internal Review Pages at the top of this issue for links you can see the build.

File: aspnetcore/mvc/views/view-compilation.md (hunk 1 above)
Comment (scottaddie) [reply aspnetcore/mvc/views/view-compilation.md]: These don't appear as tabs. We're moving away from that approach, now that we have a version selector at the top of the docs page. Only content for the selected version is displayed.

**Pull request merged**